satpy/composites/core.py

```diff
@@ -0,0 +1,535 @@
+
+        return xr.DataArray(data=data.data, attrs=new_attrs,
+                            dims=data.dims, coords=data.coords)
+
+    def _get_mode(self, attrs, num):
+        mode = attrs.get("mode")
+        if mode is None:
+            # num may not be in `self.modes` so only check if we need to
+            mode = self.modes[num]
+        return mode
+
+    def _check_datasets_and_data(self, datasets, mode):
+        datasets = self.match_data_arrays(datasets)
+        data = self._concat_datasets(datasets, mode)
+        # Skip masking if user wants it or a specific alpha channel is given.
+        if self.common_channel_mask and mode[-1] != "A":
+            data = data.where(data.notnull().all(dim="bands"))
+        # if inputs have a time coordinate that may differ slightly between
+        # themselves then find the mid time and use that as the single
+        # time coordinate value
+        time = check_times(datasets)
+        if time is not None and "time" in data.dims:
+            data["time"] = [time]
+
+        return datasets, data
+
+    def _get_updated_attrs(self, datasets, attrs, mode):
+        new_attrs = combine_metadata(*datasets)
+        # remove metadata that shouldn't make sense in a composite
+        new_attrs["wavelength"] = None
+        new_attrs.pop("units", None)
+        new_attrs.pop("calibration", None)
+        new_attrs.pop("modifiers", None)
+
+        new_attrs.update({key: val
+                          for (key, val) in attrs.items()
+                          if val is not None})
+        resolution = new_attrs.get("resolution", None)
+        new_attrs.update(self.attrs)
+        if resolution is not None:
+            new_attrs["resolution"] = resolution
+        new_attrs["sensor"] = self._get_sensors(datasets)
+        new_attrs["mode"] = mode
+
+        return new_attrs
+
+
+def check_times(projectables):
+    """Check that *projectables* have compatible times."""
+    times = []
+    for proj in projectables:
+        status = _collect_time_from_proj(times, proj)
+        if not status:
+            break
+    else:
+        return _get_average_time(times)
+
+
+def _collect_time_from_proj(times, proj):
+    status = False
+    try:
+        if proj["time"].size and proj["time"][0] != 0:
+            times.append(proj["time"][0].values)
+            status = True
+    except KeyError:
+        # the datasets don't have times
+        pass
+    except IndexError:
+        # time is a scalar
+        if proj["time"].values != 0:
+            times.append(proj["time"].values)
+            status = True
+    return status
+
+
+def _get_average_time(times):
+    # Is there a more gracious way to handle this ?
+    if np.max(times) - np.min(times) > TIME_COMPATIBILITY_TOLERANCE:
+        raise IncompatibleTimes("Times do not match.")
+    return (np.max(times) - np.min(times)) / 2 + np.min(times)
+
+
+class RGBCompositor(GenericCompositor):
```

Comment:
Maybe not in this PR, but it is probably time to remove this compositor.

Reply:
I had the same thought, but thought again and left it for the 1.0 deprecation cleanup.